header-only/tests/test_3091-layout-builder-is-valid.cpp

```diff
@@ -0,0 +1,116 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward/blob/main/LICENSE
```

Comment:
I built tests for Indexed and IndexedOption, unfortunately I couldn't figure out how to build a unit test for `Union` that made sense, so I am just relying on `is_valid` calls in the layout builder test file. I checked in the debugger and it seems to do what I expected.

Reply:
Okay, I think that's there aren't any uses of Union in Kaitai yet, so this can be something to be reevaluated later, when there's a use case.